plugins/module_utils/dcim.py

```diff
@@ -158,6 +158,10 @@ def run(self):
                     name = f"{self.module.params['parent_module_bay'].get('parent_module')} > {name}"
             elif isinstance(self.module.params["location"], dict):
                 name = f"{self.module.params['location'].get('parent', '—')} > {self.module.params['location'].get('name')} > {name}"
+        elif data.get("id"):
+            name = data["id"]
```

Comment:
Would this possibly overwrite the name variable if there is an ID and a name?

Reply:
Nope, it's all part of the same `if`/`elif` statement so it will only hit this if `name` is not there.